Cabal/Distribution/Simple/PreProcess.hs

```diff
@@ -126,20 +126,29 @@ data PreProcessor = PreProcessor {
   runPreProcessor :: (FilePath, FilePath) -- Location of the source file relative to a base dir
                   -> (FilePath, FilePath) -- Output file name, relative to an output base dir
                   -> Verbosity -- verbosity
-                  -> IO ()     -- Should exit if the preprocessor fails
+                  -> IO [FilePath] -- Should exit if the preprocessor fails
   }
 
 mkSimplePreProcessor :: (FilePath -> FilePath -> Verbosity -> IO ())
                       -> (FilePath, FilePath)
-                      -> (FilePath, FilePath) -> Verbosity -> IO ()
+                      -> (FilePath, FilePath) -> Verbosity -> IO [FilePath]
 mkSimplePreProcessor simplePP
   (inBaseDir, inRelativeFile)
+  (outBaseDir, outRelativeFile) verbosity = simplePP inFile outFile verbosity >> return []
+  where inFile  = normalise (inBaseDir  </> inRelativeFile)
+        outFile = normalise (outBaseDir </> outRelativeFile)
+
+mkLessSimplePreProcessor :: (FilePath -> FilePath -> Verbosity -> IO [FilePath])
```

Comment:
Not a very good name, IMO. Maybe add a comment explaining why it is "less simple".